src/map/status.c

```diff
@@ -8723,40 +8723,186 @@ static int status_change_start_sub(struct block_list *src, struct block_list *bl
 				//associated, and yet are not wrong/unknown. [Skotlex]
 				//break;
 				}
-			case SC_ARMOR_RESIST:
-				if (val1 > 0) {
+			case SC_ARMOR_RESIST: {
+				struct status_change_entry *sce_water = sc->data[SC_RESIST_PROPERTY_WATER];
+				struct status_change_entry *sce_ground = sc->data[SC_RESIST_PROPERTY_GROUND];
+				struct status_change_entry *sce_fire = sc->data[SC_RESIST_PROPERTY_FIRE];
+				struct status_change_entry *sce_wind = sc->data[SC_RESIST_PROPERTY_WIND];
+
+				// Water
+				int sum_water = val1 + ((sce_fire != NULL) ? sce_fire->val2 : 0);
+				bool show_icon = true;
+				if (sce_water != NULL) {
+					sum_water += sce_water->val1;
+					int left = (int)DIFF_TICK(timer->get(sce_water->timer)->tick, timer->gettick());
```

Comment:
``timer->get`` may return NULL. also sce_water->timer can be invalid timer. probably need add checks for timer field and return value of ``timer->get``?